src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -515,11 +518,16 @@ static void on_resolver_result_changed_locked(void* arg, grpc_error* error) {
         chand->exit_idle_when_lb_policy_arrives = false;
       }
       watch_lb_policy_locked(chand, chand->lb_policy.get(), state);
+    } else if (chand->resolver_result == nullptr) {
+      // Transient failure.
+      GRPC_CLOSURE_LIST_SCHED(&chand->waiting_for_resolver_result_closures);
     }
     if (!lb_policy_updated) {
       set_channel_connectivity_state_locked(
           chand, state, GRPC_ERROR_REF(state_error), "new_lb+resolver");
     }
+    grpc_channel_args_destroy(chand->resolver_result);
```

Comment:
nit: why move these lines but leave them further down than they need to be? Any reason we can't have them after line 524. `chand->resolver_result` doesn't seem to be referenced after that.

Reply:
It doesn't actually need to be reset until just before we call `NextLocked()` again.  I'm effectively changing the mode here from "keep this only until we've processed it" to "keep this for the whole callback, so that we can access it at any time".